### DISCUSSION TITLE
Hidden Pages Don't Load on Submission Edit

### DISCUSSION THREAD

**Discussion by FactorD** [team-forms/community#25] [Q&A] [answered]
1. We have a form with 6 pages. After the initial submit (Pages 1+2), 5 different users have to edit the rest of the form. 
2. The 1st editor opens the form and cannot see pages 1+2, only pages 3-4-5-6. 
3. The 1st editor edits pages 3-4-5-6.
》》All the data of the initial submit (from pages 1+2 ) is visible in submission data tab.

4. The 2nd editor enters the form. They can only see and edit page 3, all other pages are hidden to them.
》》All the data from the initial submit (P1+2) and all the 1st editors edits (P3-6) are visible in the submission data tab.

5. The 3rd editor enters the form. They can only see and edit page 4, all other pages are hidden to them.
》》All the data from the initial submit (P1+2) are visible in the submit data tab.
》》All the data submitted by the 1st and 2nd editors on page 3 are visible in the submission data tab.
》》All other data is lost and doesn't show in the submission data tab. (all data from pages 5 and 6 doesn't show)

6. The 4th editor enters the form. They can only see and edit page 5, all other pages are hidden to them.
》》All the data from the initial submit (P1+2) are visible in the submit data tab.
》》All the data submitted by the 1st, 2nd and 3rd editors on page 3 and 4 are visible in the submission data tab.
》》All other data is lost. It's not visible in the submission data tab anymore.

If the 4th editor logs in and edits page 5 before the 3rd editor has edited page 4, then all the 1st editors answers of page 4 are lost too. 

*We assume that Team Forms pages do not load their data on edit if they come AFTER the last visible page.
*Shouldn't all pages data be saved and submitted with each edit, no matter if it's visible to that particular user?

Please assist!

Best Regards,
Damon

**Comment by Erin-Dwyer** [comment]
Hi Damon,

I would be happy to assist you with this however the way you have phrased your question is a little too complex to to follow. Would you be able to provide a more simple example of the underlying issue with an example? 

Here are some points that may aid in this discussion.
1. If using the permissions tab to control who can see different section of a form then there should be no issues with component values being cleared if they are hidden. This is the recommended pattern by Team Forms that aligns with our best practise. 
2. If using advanced conditionals to hide and show components then the value of the component can get cleared when it is hidden. You shoudl check the "Clear Value When Hidden" on every component, in addition some components do not have this option and so you may need to add it manually into the JSON as shown below:

![image](https://github.com/team-forms/community/assets/154988548/b72e5185-18a5-4b53-90ea-c9e81acc66aa)

If you are using advanced conditions instead of the permissions I would recommend re-evaluating weather that is really necessary as it will make your form far more complex and difficult to maintain.

**Reply by FactorD** [reply]
Hi Erin, 

I apologize for the confusing way I phrased the question. Bottom line: Pages hidden via advanced conditionals stay even hidden from the JSON, so their data isn't submitted. If the first users fills out all 6 pages and the 2nd user can only see pages 1-3, then all the previously submitted data from the first user on pages 4, 5 and 6 is lost. 

I added "clearOnHide": false to all the pages and panels as these components don't have this option. But sadly it didn't help.

Edit 05/21: 
1. we just found out that after adding  "clearOnHide": false for pages, it is deleted after save.
2. we also found out that when the hidden page (page 2) is in the middle of two visible pages (1 and 3), the middle page data is never lost and always shows inside the JSON.

**Reply by FactorD** [reply, marked as answer]
We replaced all our conditional "show" with conditional "next" and it solved all problems.
